src/NFe/Danfe.php

```diff
@@ -1032,9 +1032,19 @@ protected function header($x = 0, $y = 0, $pag = '1', $totPag = '1')
             //endereço
             $y1     = $y1 + 5;
             $aFont  = ['font' => $this->fontePadrao, 'size' => 8, 'style' => ''];
-            $fone   = !empty($this->enderEmit->getElementsByTagName("fone")->item(0)->nodeValue)
-                ? $this->enderEmit->getElementsByTagName("fone")->item(0)->nodeValue
-                : '';
+            $fone = $this->getTagValue($this->enderEmit, "fone")!=""? $this->formatFone($this->enderEmit):'';
+            if (strlen($fone) > 0) {
+                if (strlen($fone) == 10) {
+                    $fone = $this->formatField($fone, "(##) ####-####");
+                }
+                if (strlen($fone) > 10 && substr($fone, 0, 4) != '0800') {
+                    $fone = $this->formatField($fone, "(##) #####-####");
+                } 
+        
+                if (substr($fone, 0, 4) == '0800') { 
+                    $fone = $this->formatField($fone, "#### ### ####"); 
+                }
+            }
```

Comment:
Acredito que ficaria melhor colocar essas tratativas direto na função que formata o fone, além disso, daria para tratar o primeiro IF para caso tiver tamanho 0 retornar bem no início da função, evitando o acoplamento de muitos ifs um dentro do outro.